mne/viz/topo.py

```diff
@@ -231,7 +231,7 @@ def _plot_topo(info, times, show_func, click_func=None, layout=None,
                                     unified=unified, img=img, axes=axes)
 
     # Temporarily converting the ylim to a list to avoid zip object exhaustion
-    if ylim is not None:
+    if layout.kind == 'Vectorview-all' and ylim is not None:
```

Comment:
I don't really like this. This code should be rather agnostic of the type of data. You have access to info so you should now that ylim properly used eg by using different ylim for different channel types